org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java

```diff
@@ -207,7 +207,7 @@ public void underlyingFileDeleted(IFileBuffer buffer) {
 	}
 
 	private static class LanguageServerContext {
-		boolean cancelled = false;
+		volatile boolean cancelled;
```

Comment:
Thanks for the PR. What do you think of making this an AtomicBoolean as it was before (see commit ca55022f6d7441d1b90e462675be85e8dc1275b7). I think this is more clear that working with volatiles.

Reply:
As you like, you are the code owner, IMO thee two options are equal and it is just matter of style.

Reply:
So if you want `AtomicBoolean`, just say it :-)